### PR TITLE
Adapt errors to jsonapi style

### PR DESCRIPTION
**Pull request by ShimShtein**

### CODE REVIEW

File: app/controllers/concerns/error_handling.rb [hunk 1]
```diff
@@ -3,13 +3,17 @@
...
     class BadRequest < RuntimeError; end
 
     rescue_from StandardError do |exception|
-      render json: { errors: "Server encountered an unexpected error: #{exception.inspect}" },
-             status: :internal_server_error
+      render :json => exception_hash(
+        exception,
+        status: :bad_request,
+        detail: "Server encountered an unexpected error: #{exception.inspect}"
+      ), status: :bad_request
```
Comment (bastilian) [inline app/controllers/concerns/error_handling.rb]: Shouldn't this be kept as `:internal_server_error`?

**Comment by bastilian** [comment]
Looks good to me, except for the one `bad_request`, which maybe should be kept as `:internal_server_error`. Works nicely with https://github.com/RedHatInsights/notifications-frontend/pull/58

File: app/controllers/concerns/error_handling.rb (hunk 1 above)
Comment (ShimShtein) [reply app/controllers/concerns/error_handling.rb]: My bad. Probably copy-paste error. I'll fix it.

**Comment by ShimShtein** [comment]
Rebased and fixed

**Comment by bastilian** [comment]
@ShimShtein mind rebasing once more?

**Comment by ShimShtein** [comment]
Finally! Rebased and tests are green.

**Comment by bastilian** [comment]
Awesome! Thanks @ShimShtein!

**Pull request merged**